pkg/capabilities/triggers/cron_trigger.go

```diff
@@ -0,0 +1,287 @@
+package triggers
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	// actually uses cronv3 internally for its
+	// parser to turn cron strings into cron schedules
+	"github.com/go-co-op/gocron/v2"
+	"github.com/robfig/cron/v3"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+const cronTriggerID = "cron-trigger@1.0.0"
+
+var cronTriggerInfo = capabilities.MustNewCapabilityInfo(
+	cronTriggerID,
+	capabilities.CapabilityTypeTrigger,
+	"A trigger to schedule workflow execution to run periodically at fixed times, dates, or intervals.",
+)
+
+type cronTriggerConfig struct {
+	// An identifier to register this trigger under, must be unique across all triggers within a workflow.
+	TriggerID string `json:"triggerId"`
+	// The crontab to evaluate, with second granularity.
+	// The seconds field is required. The time zone will always be automatically set to UTC.
+	Schedule string `json:"schedule"`
+}
+
+type cronTriggerInput struct{}
+
+type CronTriggerPayload struct {
+	ScheduledExecutionTime string
+}
+
+type cronJob struct {
+	ch  chan<- capabilities.CapabilityResponse
+	job gocron.Job
+}
+
+type CronTriggerService struct {
+	capabilities.CapabilityInfo
+	capabilities.Validator[cronTriggerConfig, cronTriggerInput, capabilities.CapabilityResponse]
+	jobs      map[string]cronJob
+	lggr      logger.Logger
+	mu        sync.Mutex
+	scheduler gocron.Scheduler
+	stopCh    services.StopChan
+	wg        sync.WaitGroup
+}
+
+var _ capabilities.TriggerCapability = (*CronTriggerService)(nil)
+var _ services.Service = &CronTriggerService{}
+
+func NewCronTriggerService(lggr logger.Logger) *CronTriggerService {
+	s, err := gocron.NewScheduler()
+	if err != nil {
+		return nil
+	}
+	return &CronTriggerService{
+		CapabilityInfo: cronTriggerInfo,
+		Validator:      capabilities.NewValidator[cronTriggerConfig, cronTriggerInput, capabilities.CapabilityResponse](capabilities.ValidatorArgs{Info: cronTriggerInfo}),
+		jobs:           map[string]cronJob{},
+		lggr:           logger.Named(lggr, "CronTrigger"),
+		scheduler:      s,
+		stopCh:         make(services.StopChan),
+	}
+}
+
+func (cts *CronTriggerService) RegisterTrigger(ctx context.Context, req capabilities.CapabilityRequest) (<-chan capabilities.CapabilityResponse, error) {
+	cts.mu.Lock()
+	defer cts.mu.Unlock()
+
+	config, err := cts.ValidateConfig(req.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	triggerID := getTriggerID(config.TriggerID, req.Metadata.WorkflowID)
+	if _, ok := cts.jobs[triggerID]; ok {
+		return nil, fmt.Errorf("triggerId %s already registered", triggerID)
+	}
+
+	job := gocron.CronJob(config.Schedule, true)
+
+	cronParser := cron.NewParser(
+		cron.SecondOptional | cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow,
+	)
```

Comment:
I would just use cron.NewParser(cron.WithSeconds) here so that seconds are required all the time. This should make it simpler to validate schedules upstream if we need to and will accommodate smaller granularities

Reply:
Not sure that I follow, because `cron.NewParser(cron.WithSeconds)` doesn't seem to be valid syntax --- but it doesn't matter because with all the cleanup this has gone away.